### PR TITLE
Added section about composite entities to the docs

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Add documentation about composite entities.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/nlu/entity-extraction.rst [hunk 1]
```diff
@@ -98,11 +98,82 @@ Custom Entities
...
+.. code-block:: none
+
+    I want to fly from [Berlin](city) to [San Francisco](city).
+
+However, sometimes you want to specify entities even further.
+Let's assume we want to build an assistant that should book a flight for us.
+The assistant needs to know which of the two cities in the example above is the departure city and which is the
+destination city.
+``Berlin`` and ``San Francisco`` are still cities, but they play a different role in our example.
+Composite entities allow you to assign a role and/or a group label next to the entity label.
```
Comment (erohmensing) [inline docs/nlu/entity-extraction.rst]:
```suggestion
Composite entities allow you to assign a role and/or a group label in addition to the entity label.
```

File: docs/nlu/entity-extraction.rst [hunk 2]
```diff
@@ -98,11 +98,82 @@ Custom Entities
...
+
+.. code-block:: none
+
+    I want to fly from [Berlin]{"entity": "city", "role": "departure"} to [San Francisco]{"entity": "city", "role": "destination"}.
```
Comment (erohmensing) [inline docs/nlu/entity-extraction.rst]:
```suggestion
    - I want to fly from [Berlin]{"entity": "city", "role": "departure"} to [San Francisco]{"entity": "city", "role": "destination"}.
```

File: docs/nlu/entity-extraction.rst [hunk 3]
```diff
@@ -98,11 +98,82 @@ Custom Entities
...
+
+The group label might be useful in the following situation:
+
+.. code-block:: none
+
+    Give me a [small]{"entity": "size", "group": "1"} pizza with [mushrooms]{"entity": "topping", "group": "1"} and
+    a [large]{"entity": "size", "group": "2"} [pepperoni]{"entity": "topping", "group": "2"}
```
Comment (erohmensing) [inline docs/nlu/entity-extraction.rst]: Is the "group" here to reference which topping goes with which pizza? it's not immediately obvious to me.

File: docs/nlu/entity-extraction.rst [hunk 4]
```diff
@@ -98,11 +98,82 @@ Custom Entities
...
+    Composite entities are currently only supported by the ``DIETClassifier``.
+
+In order to properly train your model with composite entities, make sure to include enough training data examples
+for every combination of entity and role/group label.
+Also make sure to have some variations in your training data, so that the model is able to generalize.
+For example, you should not only have example like ``fly FROM x TO y``, but also include examples like
+``fly TO y FROM x``.
+
+To fill slots from entities with a specific role/group, you need to either use forms or use a custom action.
+
```
Comment (erohmensing) [inline docs/nlu/entity-extraction.rst]:
```suggestion


```

File: docs/nlu/entity-extraction.rst (hunk 3 above)
Comment (tabergma) [reply docs/nlu/entity-extraction.rst]: Yes, will add a sentence.

**Comment by tabergma** [comment]
`CRFEntityExtractor` is not deprecated.

**Pull request merged**